cpp/scenario/gazebo/include/scenario/gazebo/Joint.h

```diff
@@ -0,0 +1,185 @@
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SCENARIO_GAZEBO_JOINT_H
+#define SCENARIO_GAZEBO_JOINT_H
+
+#include <ignition/gazebo/Entity.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/EventManager.hh>
+
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace scenario {
+    namespace base {
+        struct PID;
+        struct Limit;
+        enum class JointType
+        {
+            Invalid,
+            Fixed,
+            Revolute,
+            Prismatic,
+            Ball,
+        };
+        enum class JointControlMode
+        {
+            Idle,
+            Force,
+            Velocity,
+            Position,
+            PositionInterpolated,
+        };
+    } // namespace base
+    namespace gazebo {
+        class Joint;
+        class Model;
+    } // namespace gazebo
+} // namespace scenario
+
+class scenario::gazebo::Joint
+{
+public:
+    Joint();
+    virtual ~Joint();
+
+    // ============
+    // Gazebo Joint
+    // ============
+
+    uint64_t id() const;
+    bool initialize(const ignition::gazebo::Entity jointEntity,
+                    ignition::gazebo::EntityComponentManager* ecm,
+                    ignition::gazebo::EventManager* eventManager);
+    bool createECMResources();
+
+    bool historyOfAppliedJointForcesEnabled() const;
+    bool enableHistoryOfAppliedJointForces( //
+        const bool enable,
+        const size_t maxHistorySize = 100);
+    std::vector<double> getHistoryOfAppliedJointForces() const;
+
+    // ==========
+    // Joint Core
+    // ==========
+
+    size_t dofs() const;
+    std::string name() const;
+    base::JointType type() const;
+
+    base::Limit positionLimit() const;
+
+    base::JointControlMode controlMode() const;
+    bool setControlMode(const base::JointControlMode mode);
+
+    double controllerPeriod() const;
+
+    double maxEffort() const;
+    bool setMaxEffort(const double maxEffort);
+
+    base::PID pid() const;
+    bool setPID(const base::PID& pid);
+
+    // ==================
+    // Single DOF methods
+    // ==================
+
+    double force(const size_t dof = 0) const;
```

Comment:
Besides the discussion above, I think that the initial `force` and the proposed `forceTarget` are not expressive enough names.
From the user's perspective, it might be better to have a longer and more espressive name, e.g. something like `getForceReference` or `getSentForceReference`.

Reply:
Refer to https://github.com/robotology/gym-ignition/pull/158#discussion_r395493760